### PR TITLE
improve arguments for ApplicationContextCommand

### PR DESCRIPTION
**Pull request by yamkazu**
This makes it possible to pass command line arguments to `ApplicationContextCommand`.

If there is a Gradle task as following:

```
project.tasks.create("myTask", ApplicationContextCommandTask) {
    classpath = project.sourceSets.main.runtimeClasspath + project.configurations.console
    command = 'my-task'
    if (project.hasProperty('args')) {
        args(project.args.split(' '))
    }
}
```

and execute the task with arguments:

```
grails my-task args1 args2
```

The above will pass `['args1', 'args2']` to the implementation of `ApplicationContextCommand`.

### CODE REVIEW

File: grails-shell/src/main/groovy/org/grails/cli/gradle/GradleInvoker.groovy [hunk 1]
```diff
@@ -58,6 +58,9 @@ class GradleInvoker {
                 arguments << '--stacktrace'
                 arguments << '-Dgrails.full.stacktrace=true'
             }
+            if (commandLine.remainingArgs || commandLine.undeclaredOptions) {
+                arguments << "-Pargs=${commandLine.remainingArgsWithOptionsString}".toString()
```
Comment (graemerocher) [inline grails-shell/src/main/groovy/org/grails/cli/gradle/GradleInvoker.groovy]: Is this change necessary? Seems requiring all commands that use GradeInvoker to pass -Pargs= is overkill

I think it would be better to change:

https://github.com/grails/grails-core/blob/master/grails-shell/src/main/groovy/org/grails/cli/gradle/commands/GradleTaskCommandAdapter.groovy#L55

To pass the arguments only when invoking the command

**Comment by yamkazu** [comment]
@graemerocher thanks for your comment. I fixed it.
